cmd/k8s-operator/svc.go

```diff
@@ -158,7 +164,13 @@ func (a *ServiceReconciler) maybeProvision(ctx context.Context, logger *zap.Suga
 		{Hostname: tsHost},
 	}
 	for _, ip := range tsIPs {
-		ingress = append(ingress, corev1.LoadBalancerIngress{IP: ip})
+		addr, err := netip.ParseAddr(ip)
+		if err != nil {
+			continue
+		}
+		if (addr.Is4() && clusterIPAddr.Is4()) || (addr.Is6() && clusterIPAddr.Is6()) {
```

Comment:
If both are valid you should be able to `addr.Is4() == other.Is4()` here as a simplification

Reply:
fair, done